internal/extender/resource.go

```diff
@@ -275,8 +275,9 @@ func (s *SparkSchedulerExtender) selectDriverNode(ctx context.Context, driver *v
 			svc1log.SafeParam("nodeNames", nodeNames))
 		return driverReservedNode, success, nil
 	}
-	availableNodes, err := utils.ListWithPredicate(s.nodeLister, func(node *v1.Node) bool {
-		return helper.PodMatchesNodeSelectorAndAffinityTerms(driver, node)
```

Comment:
`PodMatchesNodeSelectorAndAffinityTerms` func is deprecated. Used the same approach as in k8s PR https://github.com/kubernetes/kubernetes/pull/99959/files